medicines/doc-index-updater/src/search_client/mod.rs

```diff
@@ -0,0 +1,183 @@
+use serde_derive::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct AzureHighlight {
+    #[serde(rename = "content")]
+    content: Vec<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AzureResult {
+    pub doc_type: String,
+    pub file_name: String,
+    pub metadata_storage_name: String,
+    pub metadata_storage_path: String,
+    pub product_name: Option<String>,
+    pub substance_name: Vec<String>,
+    pub title: String,
+    pub created: String,
+    pub facets: Vec<String>,
+    pub keywords: Option<String>,
+    pub metadata_storage_size: i32,
+    pub release_state: String,
+    pub rev_label: Option<String>,
+    pub suggestions: Vec<String>,
+    #[serde(rename = "@search.score")]
+    pub score: f32,
+    #[serde(rename = "@search.highlights")]
+    pub highlights: Option<AzureHighlight>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AzureSearchResults {
+    #[serde(rename = "value")]
+    pub search_results: Vec<AzureResult>,
+    #[serde(rename = "@odata.context")]
+    context: String,
+    #[serde(rename = "@odata.count")]
+    count: Option<i32>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AzureIndexChangedResults {
+    pub value: Vec<AzureIndexChangedResult>,
+    #[serde(rename = "@odata.context")]
+    context: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AzureIndexChangedResult {
+    pub key: String,
+    pub status: bool,
+    #[serde(rename = "errorMessage")]
+    pub error_message: Option<String>,
+    #[serde(rename = "statusCode")]
+    pub status_code: u16,
+}
+
+#[derive(Clone)]
+struct AzureConfig {
+    search_service: String,
+    search_index: String,
+    api_key: String,
+    api_version: String,
+}
+
+pub struct AzureSearchClient {
+    client: reqwest::Client,
+    config: AzureConfig,
+}
+
+pub fn get_env(key: &str) -> String {
+    std::env::var(key).expect(&format!("Set env variable {} first!", key))
+}
+
+pub fn factory() -> AzureSearchClient {
+    let api_key = get_env("API_ADMIN_KEY");
+    let search_index = get_env("AZURE_SEARCH_INDEX");
+    let search_service = get_env("SEARCH_SERVICE");
+    let api_version = get_env("AZURE_SEARCH_API_VERSION");
```

Comment:
can these be added to the `.env.example`? that's the most complete version of the variables we need

Reply:
Ah, that's an excellent point. +:100: